src/Service.gs

```diff
@@ -497,69 +503,40 @@ Service_.prototype.refresh = function() {
 };
 
 /**
- * Saves a token to the service's property store and cache.
- * @param {Object} token The token to save.
- * @private
+ * Gets the storage layer for this service, used to persist tokens.
+ * Custom values associated with the service can be stored here as well.
+ * The key <code>null</code> is used to to store the token and should not
+ * be used.
+ * @return {Storage} The service's storage.
  */
-Service_.prototype.saveToken_ = function(token) {
+Service_.prototype.getStorage = function() {
   validate_({
     'Property store': this.propertyStore_
   });
-  var key = this.getPropertyKey_();
-  var value = JSON.stringify(token);
-  this.propertyStore_.setProperty(key, value);
-  if (this.cache_) {
-    this.cache_.put(key, value, 21600);
+  if (!this.storage_) {
+    var prefix = 'oauth2.' + this.serviceName_;
+    this.storage_ = new Storage(prefix, this.propertyStore_, this.cache_);
   }
-  this.token_ = token;
+  return this.storage_;
 };
 
 /**
- * Gets the token from the service's property store or cache.
- * @return {Object} The token, or null if no token was found.
+ * Saves a token to the service's property store and cache.
+ * @param {Object} token The token to save.
+ * @private
  */
-Service_.prototype.getToken = function() {
-  validate_({
-    'Property store': this.propertyStore_
-  });
-
-  // Check in-memory cache.
-  if (this.token_) {
-    return this.token_;
-  }
-
-  var key = this.getPropertyKey_();
-  var token;
-
-  // Check CacheService cache.
-  if (this.cache_ && (token = this.cache_.get(key))) {
-    token = JSON.parse(token);
-    this.token_ = token;
-    return token;
-  }
-
-  // Check PropertiesService store.
-  if ((token = this.propertyStore_.getProperty(key))) {
-    if (this.cache_) {
-      this.cache_.put(key, token, 21600);
-    }
-    token = JSON.parse(token);
-    this.token_ = token;
-    return token;
-  }
-
-  // Not found.
-  return null;
+Service_.prototype.saveToken_ = function(token) {
+  var storage = this.getStorage();
+  return storage.setValue(null, token);
 };
 
 /**
- * Generates the property key for a given service name.
- * @param {string} serviceName The name of the service.
- * @return {string} The property key.
- * @private
+ * Gets the token from the service's property store or cache.
+ * @return {Object} The token, or null if no token was found.
  */
-Service_.prototype.getPropertyKey_ = function() {
-  return 'oauth2.' + this.serviceName_;
+Service_.prototype.getToken = function() {
+  var storage = this.getStorage();
+  return storage.getValue(null);
```

Comment:
Don't declare a `var` if we can prevent it. Chain when possible.
`return this.getStorage().getValue(null);`

Reply:
Done.